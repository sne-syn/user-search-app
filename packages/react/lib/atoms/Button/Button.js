import React from 'react';

const Button = ({ label }) => {
    return (React.createElement("button", { className: "dse-button__container", type: "submit" }, label));
};

export { Button as default };
//# sourceMappingURL=Button.js.map
