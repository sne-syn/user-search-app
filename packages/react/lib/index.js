export { default as Avatar } from './atoms/Avatar/Avatar.js';
export { default as Button } from './atoms/Button/Button.js';
export { default as CardSubtitle } from './atoms/CardSubtitle/CardSubtitle.js';
export { default as CardTitle } from './atoms/CardTitle/CardTitle.js';
export { default as Icon } from './atoms/Icon/Icon.js';
export { default as Input } from './atoms/Input/Input.js';
export { default as Logo } from './atoms/Logo/Logo.js';
export { default as DateInfo } from './atoms/DateInfo/DateInfo.js';
export { default as DefinitionList } from './atoms/DefinitionList/DefinitionList.js';
export { default as ErrorMessage } from './atoms/ErrorMessage/ErrorMessage.js';
export { default as TextBlock } from './atoms/TextBlock/TextBlock.js';
export { default as ThemeToggler } from './atoms/ThemeToggler/ThemeToggler.js';
export { default as VisuallyHidden } from './atoms/VisuallyHidden/VisuallyHidden.js';
export { default as Wrapper } from './atoms/Wrapper/Wrapper.js';
//# sourceMappingURL=index.js.map
