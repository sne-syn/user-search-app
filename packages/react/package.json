{
  "name": "@user-search-app/react",
  "version": "1.0.0",
  "main": "lib/index.js",
  "license": "MIT",
  "scripts": {
    "build": "rollup -c",
    "dev": "yarn build --watch",
    "test": "jest --verbose",
    "test:watch": "yarn test --watch"
  },
  "devDependencies": {
    "@babel/preset-env": "^7.18.10",
    "@babel/preset-react": "^7.18.6",
    "@babel/preset-typescript": "^7.18.6",
    "@rollup/plugin-image": "^2.1.1",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^13.3.0",
    "@types/jest": "^28.1.6",
    "@types/react": "^18.0.15",
    "jest": "^28.1.3",
    "jest-environment-jsdom": "^28.1.3",
    "react": "^18.2.0",
    "rollup": "^2.77.0",
    "rollup-plugin-img": "^1.1.0",
    "rollup-plugin-typescript2": "^0.32.1",
    "typescript": "^4.7.4"
  }
}
