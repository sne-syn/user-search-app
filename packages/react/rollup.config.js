import Ts from "rollup-plugin-typescript2";
import image from "@rollup/plugin-image";

export default {
  input: [
    "src/index.ts",
    "src/atoms/Avatar/index.ts",
    "src/atoms/Button/index.ts",
    "src/atoms/CardSubtitle/index.ts",
    "src/atoms/CardTitle/index.ts",
    "src/atoms/Icon/index.ts",
    "src/atoms/Input/index.ts",
    "src/atoms/Link/index.ts",
    "src/atoms/DefinitionList/index.ts",
    "src/atoms/DateInfo/index.ts",
    "src/atoms/Logo/index.ts",
    "src/atoms/TextBlock/index.ts",
    "src/atoms/ThemeToggler/index.ts",
    "src/atoms/VisuallyHidden/index.ts",
    "src/atoms/Wrapper/index.ts",
  ],
  output: {
    dir: "lib",
    format: "esm",
    sourcemap: true,
  },
  plugins: [
    Ts(),
    image({
      extensions: /\.(png|jpg|jpeg|gif|svg)$/,
      limit: 10000,
    }),
  ],
  preserveModules: true,
  external: ["react", "@ds.e/foundation"],
};
