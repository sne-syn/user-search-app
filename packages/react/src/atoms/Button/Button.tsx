import React, { FC } from "react";

interface ButtonProps {
  label: string;
  type?: "submit" | "button";
  disabled?: boolean;
  style?: React.CSSProperties;
  handleClick?: () => void;
}

const Button: FC<ButtonProps> = ({
  label,
  style,
  handleClick,
  disabled = false,
  type = "button",
}) => {
  return (
    <button
      style={style}
      className="dse-button__container"
      type={type}
      onClick={() => handleClick && handleClick()}
      disabled={disabled}
    >
      {label}
    </button>
  );
};

export default Button;
