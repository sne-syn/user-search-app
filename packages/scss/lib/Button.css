@import url("https://fonts.googleapis.com/css2?family=Space+Mono:wght@400;700&display=swap");
/*=============================================
    #Variables
=============================================*/
/**
 * TABLE OF CONTENTS
 * 
 * COLORS
 * Brand colors...................These are the main brand colors. They will be used the most
 * Neutral colors.................White and black, gray scale, secondary colors.
 * Utility colors.................Other colors to represent states: info, warning, success, etc.
 * 
 * TYPOGRAPHY
 * Font families..................The fonts used in the design system.
 * Font weights...................Font weights
 * Sizes..........................Font sizing for body text, paragraphs, inputs
 * Heading sizes..................Font sizing for page headings
 *
 * BREAKPOINTS
 * Breakpoints....................Global responsive breakpoints definitions.
 *
 * SPACING
 * Spacing........................Global fixed set of spacing variables
 */
/*=============================================
=                  Colors                     =
=============================================*/
/* Brand Colors */
/* Neutral Colors */
/* Utility colors */
/*=====  End of Colors  ======*/
/*=============================================
=                Typography                   =
=============================================*/
/* Font families */
/**
The Space Mono font. We'll use this for the body text.
 *     https://fonts.google.com/specimen/Space+Mono
 */
/* Font sizes */
/* Heading font sizes */
/* Font weights */
/* Line Height */
/*=====  End of Typography  ======*/
/*=============================================
=            Breakpoints                      =
=============================================*/
/*=====  End of Breakpoints  ======*/
/*=============================================
=            Spacing                          =
=============================================*/
/*=====  End of Spacing  ======*/
/*=============================================
=            Foundation - colors              =
=============================================*/
/**
 * This file defines the actual colors that will be used for styling. They will default to the palette
 * we defined in the _variable.scss file. This is our default palette, and devs can override this
 * with their own variables.
 */
/*=============================================
=            Themed colors               =
=============================================*/
/* Dark Theme */
/* Light Theme */
/*=============================================
=                 Shared                     =
=============================================*/
/*=============================================
=        Foundation - Typography              =
=============================================*/
/**
 * This file defines the default font sizes to be used on headings and texts. The defaults come
 * from the palette defined in _variables.scss.
 */
/*=============================================
=            Body font family                  =
=============================================*/
/*=============================================
=            Body font sizes                  =
=============================================*/
/*=============================================
=            Heading font sizes               =
=============================================*/
/*=============================================
=            Head font weights                =
=============================================*/
/*=============================================
=            Mixins                           =
=============================================*/
/**
 *
 * Helpful reusable function-like things that will be used to implement
 * styles for components.
 *
 */
html {
  box-sizing: border-box;
}

*,
*::before,
*::after {
  box-sizing: inherit;
}

body {
  background-color: var(--dse-body-bg-color, #f5f7ff);
  box-sizing: inherit;
  font-family: var(--dse-body-font-family, "Space Mono", "Courier New", Courier, monospace);
  font-style: normal;
  font-weight: 400;
}

.dse-button__container {
  font-size: 1rem;
  line-height: 2.375rem;
  background: var(--dse-btn-primary-bg, #0077ff);
  border: none;
  border-radius: 10px;
  color: var(--dse-btn-primary-color, white);
  cursor: pointer;
  display: flex;
  font-family: var(--dse-body-font-family, "Space Mono", "Courier New", Courier, monospace);
  font-style: normal;
  font-weight: 700;
  justify-content: center;
  min-width: 106px;
  padding: 5px 0.938rem;
}

.dse-button__container:hover, .dse-button__container:disabled {
  background: var(--dse-btn-primary-bg-hover, #61abff);
}

.dse-button__container:disabled {
  cursor: default;
}
