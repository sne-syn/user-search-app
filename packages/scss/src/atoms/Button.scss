@import "foundation/all";

.dse-button__container {
  @include font($fs-xs, $lh-base);
  @include minSize($btn-width, $btn-height);

  align-items: center;
  background: $btn-primary-bg;
  border: none;
  border-radius: $border-radius-btn;
  color: $btn-primary-color;
  cursor: pointer;
  display: flex;
  font-family: inherit;
  font-weight: $fw-bold;
  justify-content: center;
  padding: map-get($spacing, "sm");

  &:hover,
  &:disabled {
    background: $btn-primary-bg-hover;
  }

  &:disabled {
    cursor: default;
  }

  @include tablet {
    @include font($fs-base, $lh-xs);
    @include minSize($btn-width-tablet, $btn-height-tablet);
  }
}
