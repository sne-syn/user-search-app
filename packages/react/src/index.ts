import Avatar from "./atoms/Avatar";
import Button from "./atoms/Button";
import CardSubtitle from "./atoms/CardSubtitle";
import CardTitle from "./atoms/CardTitle";
import Input from "./atoms/Input";
import ListContent from "./atoms/ListContent";
import ListIcon from "./atoms/ListIcon";
import Logo from "./atoms/Logo";
import DateInfo from "./atoms/DateInfo";
import DefinitionList from "./atoms/DefinitionList";
import ErrorMessage from "./atoms/ErrorMessage";
import TextBlock from "./atoms/TextBlock";
import ThemeToggler from "./atoms/ThemeToggler";
import VisuallyHidden from "./atoms/VisuallyHidden";
import Wrapper from "./atoms/Wrapper";

export {
  Avatar,
  Button,
  CardSubtitle,
  CardTitle,
  Input,
  ListContent,
  ListIcon,
  Logo,
  DateInfo,
  DefinitionList,
  ErrorMessage,
  TextBlock,
  ThemeToggler,
  VisuallyHidden,
  Wrapper,
};
