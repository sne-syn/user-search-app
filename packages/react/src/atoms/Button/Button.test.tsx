import React from "react";
import Button from "./Button";
import { render, screen, fireEvent } from "@testing-library/react";
const handleClick = jest.fn();

describe("Button", () => {
  test("was rendered", () => {
    render(<Button label="Send" />);
    expect(screen.getByRole("button")).toBeInTheDocument;
    expect(screen.getByRole("button")).toMatchSnapshot();
  });

  test("was rendered with style provided", () => {
    render(<Button label="Send" style={{ width: "100px" }} />);
    expect(screen.getByRole("button")).toHaveStyle("width: 100px");
    expect(screen.getByRole("button")).toMatchSnapshot();
  });

  test("triggers onClick function by click", () => {
    render(<Button label="Send" handleClick={handleClick} />);
    fireEvent.click(screen.getByRole("button"));
    expect(handleClick).toBeCalledTimes(1);
  });

  test("is disabled", () => {
    render(<Button label="Send" handleClick={handleClick} disabled={true} />);
    expect(screen.getByRole("button")).toBeDisabled();
    expect(screen.getByRole("button")).toMatchSnapshot();
  });
});
