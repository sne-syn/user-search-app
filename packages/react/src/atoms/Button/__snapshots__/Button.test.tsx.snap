// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button is disabled 1`] = `
<button
  class="dse-button__container"
  disabled=""
  type="button"
>
  Send
</button>
`;

exports[`Button was rendered 1`] = `
<button
  class="dse-button__container"
  type="button"
>
  Send
</button>
`;

exports[`Button was rendered with style provided 1`] = `
<button
  class="dse-button__container"
  style="width: 100px;"
  type="button"
>
  Send
</button>
`;
